socorro/external/postgresql/dbapi2_util.py

```diff
@@ -15,6 +17,49 @@ class SQLDidNotReturnSingleRow (Exception):
     pass
 
 
+#==============================================================================
+class FetchAllSequence(Sequence):
+    """A sequence wrapper for results of a PG cursor's fetchall()
+    that when supplied with the cursor description gives you an
+    iterable that works the same as regular fetchall() but
+    also offers the convenience method zipped().
+
+    You can use this like this::
+
+        >>> sql = "select col1, col2 from some_table"
+        >>> cursor = connection.cursor()
+        >>> cursor.excute(sql, ())
+        >>> seq = FetchAllSequence(cursor.fetchall(), cursor.description)
+        >>> seq.zipped()
+        [{'col1': value1, 'col2': value2}, {'col1': valueA, 'col2': valueB}]
+
+    """
+
+    def __init__(self, rows, description):
+        self.rows = rows
+        self.description = description
+
+    def __getitem__(self, index):
+        return self.rows[index]
+
+    def __len__(self):
+        return len(self.rows)
+
+    def __contains__(self, value):
+        return value in self.rows
+
+    def __iter__(self):
+        for x in self.rows:
+            yield x
+
+    def __str__(self):
+        return str(self.rows)
+
+    def zipped(self):
```

Comment:
I actually like `zipped`. It's short and means something. `iterable_with_column_names` also means something but it's much verbose. I guess I get first dibs here because I wrote the patch :)

Regarding the override-ability of setting the `mapping_class`. Perhaps it's just me but I don't like adding code just because it's doable. I'd rather add it the day we need it. After all, our tooling (github, automatic deployment etc) is getting better. 

However, should it not **always** be `DotDict` instead of dict? DotDict isn't any slower because it's based on `dict` so the only "excess" `DotDict` has only comes in when you do the `__getitem__`. 
Actually, I think that looks like quite fancy. To be able to do...:
```python
results = self.query(sql)
for each in results.zipped():
    print each.first_name, each.last_name
```
That's after all what you'd expect from any abstraction.